ytfzf

```diff
@@ -571,7 +571,27 @@ set_vars () {
     #this comes from invidious' api
     thumbnail_quality="high"
     sub_link_count="2"
-    invidious_instance="https://y.com.sb" yt_video_link_domain="https://youtube.com"
+
+    # URL for instance list sorted by type, health and api (ones without API do not work!)
+    instances_url="https://api.invidious.io/instances.json?sort_by=type,health,api" 
+
+    # Cache location.
+    instances_file="$cache_dir/instances.json"
+
+    # If file doesn't already exist (or if force-refresh is requested), cURL and cache it.
+    # TODO: implement `--refresh-inv-instances` to force-refresh this list
+    # TODO: + update docs
+    [ ! -f "$instances_file" ] && curl -X GET -s "$instances_url" > "$instances_file"
+ 
+    # The pipeline does the following:
+    #   - Get top 10 instance URIs
+    #   - Output these on individual lines
+    #   - Shuffle the list
+    #   - Choose the first entry
+    invidious_instance=$(jq -r 'map(.[1] | .uri) | .[:10] | join("\n")' < "$instances_file" \
+                      | shuf | head -n 1)
```

Comment:
I think this should be moved to the `Post opt scrape` section around line 2293.
If the user sets `instances_file` or `cache_dir` in their config, the file will be in the wrong place because the config is read after creating the file.

Then, instead of just setting `invidious_instance` we should only set it if it's not already set, so
```sh
: "${invidious_instance:=$(jq -r 'map(.[1] | .uri) | .[:10] | join("\n")' < "$instances_file" \
                      | shuf | head -n 1)}"
```

Reply:
I believe it is now in the proper section. Can you confirm if this is what you meant?

Reply:
All good, just change:

```sh
instances_file="$cache_dir/instances.json"
```
to
```sh
: "${instances_file:="$cache_dir/instances.json"}"
```

Reply:
Ah, yes. Fixed now.

Also, would you prefer to isolate updates for `ytfzf(1)` required to document `--refresh-inv-instances` in a separate PR, or should we club it in along with this PR?

Reply:
Lets just do it in this PR.